index.js

```diff
@@ -12,11 +14,34 @@ exports.lintText = (str, opts) => {
 		delete opts.overrides;
 
 		const filename = path.relative(opts.cwd, opts.filename);
+
 		const foundOverrides = optionsManager.findApplicableOverrides(filename, overrides);
 		opts = optionsManager.mergeApplicableOverrides(opts, foundOverrides.applicable);
 	}
 
 	opts = optionsManager.buildConfig(opts);
+	const defaultIgnores = optionsManager.getIgnores({}).ignores;
+
+	if (opts.ignores && !arrayEqual(defaultIgnores, opts.ignores) && typeof opts.filename !== 'string') {
```

Comment:
- [x] `array-equal` => `lodash.isequal`
- the check is there to test if the user provided ignores via `.ignores` or `.overrides[index].ignores`. `opts.ignores` is always defined